### ISSUE TITLE
Coredump during end of space (assert tripped in ~mutation_fragment_stream_validator)

### ISSUE THREAD

**Issue by juliayakovlev** [scylladb/scylla#5407]
*Installation details*
Scylla version (or git commit hash): 666.development-0.20191021.765dc86de43
Cluster size: 6 nodes (i3.4xlarge)
OS (RHEL/CentOS/Ubuntu/AWS AMI):
Found during longevity-50gb-4days

Happened during enospc nemesis (it happend 4 times on the different nodes).

Coredump:
```
PID: 88072 (scylla)
           UID: 996 (scylla)
           GID: 1001 (scylla)
        Signal: 6 (ABRT)
     Timestamp: Thu 2019-10-24 14:15:16 UTC (2min 8s ago)
  Command Line: /usr/bin/scylla --blocked-reactor-notify-ms 500 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --network-stack posix --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 1-7,9-15
    Executable: /opt/scylladb/libexec/scylla
 Control Group: /scylla.slice/scylla-server.slice/scylla-server.service
          Unit: scylla-server.service
         Slice: scylla-server.slice
       Boot ID: a2d470f51fc5489f95c339b9ff52621b
    Machine ID: df877a200226bc47d06f26dae0736ec9
      Hostname: ip-10-0-114-94.eu-west-1.compute.internal
      Coredump: /var/lib/systemd/coredump/core.scylla.996.a2d470f51fc5489f95c339b9ff52621b.88072.1571926516000000
       Message: Process 88072 (scylla) of user 996 dumped core.
                
                Stack trace of thread 88072:
                #0  0x00007fa01d7a6e75 raise (libc.so.6)
                #1  0x00007fa01d79195e abort (libc.so.6)
                #2  0x00007fa01d791769 __assert_fail_base.cold (libc.so.6)
                #3  0x00007fa01d79f566 __assert_fail (libc.so.6)
                #4  0x000000000298051f _ZN5boost9container6vectorIPN7seastar8internal9linux_abi4iocbENS0_3dtl24static_storage_allocatorIS6_Lm1024EEEvE4backEv (scylla)
                #5  0x00000000029ef89f _ZN7seastar7reactor9fdatasyncEi (scylla)
                #6  0x000000000290de83 _ZN7seastar15posix_file_impl5flushEv (scylla)
                #7  0x00000000029c5005 _ZN7seastar4file5flushEv (scylla)
                #8  0x00000000029c5766 operator() (scylla)
                #9  0x00000000029810e2 _ZN7seastar7reactor9run_tasksERNS0_10task_queueE (scylla)
                #10 0x00000000029812f0 _ZN7seastar7reactor14run_some_tasksEv (scylla)
                #11 0x00000000029fced6 _ZN7seastar7reactor14run_some_tasksEv (scylla)
                #12 0x000000000292e4a2 _ZN7seastar12app_template14run_deprecatedEiPPcOSt8functionIFvvEE (scylla)
                #13 0x000000000292f0bf _ZN7seastar12app_template3runEiPPcOSt8functionIFNS_6futureIJiEEEvEE (scylla)
                #14 0x00000000007818b2 main (scylla)
                #15 0x00007fa01d792f33 __libc_start_main (libc.so.6)
                #16 0x00000000007cdcde _start (scylla)
                
                Stack trace of thread 88079:
                #0  0x0000000000e81156 _ZN8sstables2mc6writer24consume_end_of_partitionEv (scylla)
                #1  0x0000000000f2aae0 _ZN20flat_mutation_reader4impl26consume_pausable_in_threadISt17reference_wrapperINS0_16consumer_adapterI35stable_flattened_mutations_consumerI22compact_for_compactionIN8sstables25compacting_sstable_writerE33noop_compacted_fragments_consumerEEEEENS_6filterEEEvT_T0_NSt6chrono10ti
me_pointIN7seastar12lowres_clockENSG_8durationIlSt5ratioILl1ELl1000EEEEEE (scylla)
                #2  0x0000000000f2be8e _ZZN8sstables10compaction3runI33noop_compacted_fragments_consumerEEN7seastar6futureIJNS_15compaction_infoEEEESt10unique_ptrIS0_St14default_deleteIS0_EET_ENUlvE_clEv (scylla)
                #3  0x0000000000f2c5b8 _ZN7seastar20noncopyable_functionIFvvEE17direct_vtable_forIZZNS_5asyncIZN8sstables10compaction3runI33noop_compacted_fragments_consumerEENS_6futureIJNS5_15compaction_infoEEEESt10unique_ptrIS6_St14default_deleteIS6_EET_EUlvE_JEEENS_8futurizeINSt9result_ofIFNSt5decayISG_E4typeEDpN
SK_IT0_E4typeEEE4typeEE4typeENS_17thread_attributesEOSG_DpOSN_ENUlRZNS4_ISH_JEEESV_SW_SX_SZ_E4workE_clES11_EUlvE_E4callEPKS2_ (scylla)
                #4  0x00000000007cb8b4 _ZNK7seastar20noncopyable_functionIFvvEEclEv (scylla)
                
                Stack trace of thread 88084:
                #0  0x0000000000ea8602 _ZN13data_consumer24continuous_data_consumerIN8sstables27data_consume_rows_context_mEE9read_vintI13unsigned_vintLNS3_8prestateE7ELS6_9EmEENS3_11read_statusERN7seastar16temporary_bufferIcEERT2_ (scylla)
                #1  0x0000000000ebe54f _ZN8sstables27data_consume_rows_context_m16do_process_stateERN7seastar16temporary_bufferIcEE (scylla)
                #2  0x0000000000ebfbcb _ZN7seastar6repeatIZNS_12input_streamIcE7consumeISt17reference_wrapperIN8sstables27data_consume_rows_context_mEEEENS_6futureIJEEEOT_EUlvE_EES9_SA_ (scylla)
                #3  0x0000000000ef29b0 _ZZZN8sstables23sstable_mutation_readerINS_27data_consume_rows_context_mENS_17mp_row_consumer_mEE11fill_bufferENSt6chrono10time_pointIN7seastar12lowres_clockENS4_8durationIlSt5ratioILl1ELl1000EEEEEEENKUlvE1_clEvENKUlvE0_clEv (scylla)
                #4  0x0000000000ef8eb0 _ZN7seastar8do_untilIZN8sstables23sstable_mutation_readerINS1_27data_consume_rows_context_mENS1_17mp_row_consumer_mEE11fill_bufferENSt6chrono10time_pointINS_12lowres_clockENS6_8durationIlSt5ratioILl1ELl1000EEEEEEEUlvE1_ZNS5_11fill_bufferESD_EUlvE0_EENS_6futureIJEEET0_T_ (scylla
)
                #5  0x0000000000efa0b5 _ZN8sstables23sstable_mutation_readerINS_27data_consume_rows_context_mENS_17mp_row_consumer_mEE11fill_bufferENSt6chrono10time_pointIN7seastar12lowres_clockENS4_8durationIlSt5ratioILl1ELl1000EEEEEE (scylla)
                #6  0x0000000000c8c92d _ZN7seastar17parallel_for_eachIPN22mutation_reader_merger29reader_and_last_fragment_kindEZNS1_12prepare_nextENSt6chrono10time_pointINS_12lowres_clockENS4_8durationIlSt5ratioILl1ELl1000EEEEEEEUlS2_E_EENS_6futureIJEEET_SF_OT0_ (scylla)
```

Aborting on shard backtrace:

```
0x0000000002a1ffd2
0x00000000029849ab
0x0000000002984ca5
0x0000000002984d40
0x00007f17818abe7f
/opt/scylladb/libreloc/libc.so.6+0x0000000000037e74
/opt/scylladb/libreloc/libc.so.6+0x0000000000022894
/opt/scylladb/libreloc/libstdc++.so.6+0x000000000009e6b5
/opt/scylladb/libreloc/libstdc++.so.6+0x00000000000aa86b
/opt/scylladb/libreloc/libstdc++.so.6+0x00000000000a9878
/opt/scylladb/libreloc/libstdc++.so.6+0x00000000000aa223
0x00007f1781066c62
0x00007f17810671b0
/opt/scylladb/libreloc/libstdc++.so.6+0x00000000000aab1b
0x00000000005551f7
0x0000000000cbbe2b
0x000000000048614f
0x0000000000d5cc68
0x00000000007cb8b3
```

Decoded:
```
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /usr/include/c++/9/ext/new_allocator.h:105
seastar::print_with_backtrace(seastar::backtrace_buffer&) at /usr/include/c++/9/ext/new_allocator.h:105
seastar::print_with_backtrace(char const*) at /usr/include/c++/9/ext/new_allocator.h:105
void seastar::install_oneshot_signal_handler<6, &seastar::sigabrt_action>()::{lambda(int, siginfo_t*, void*)#1}::_FUN(int, siginfo_t*, void*) at /usr/include/c++/9/ext/new_allocator.h:105
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0
?? ??:0
on_internal_error(seastar::logger&, seastar::basic_sstring<char, unsigned int, 15u, true> const&) [clone .cold] at /usr/include/c++/9/ext/new_allocator.h:105
mutation_fragment_stream_validator::~mutation_fragment_stream_validator() at /usr/include/c++/9/ext/new_allocator.h:105
addr2line: Dwarf Error: Offset (14744489) greater than or equal to .debug_info size (6062233).
addr2line: Dwarf Error: Unable to read alt ref 14744489.
sstables::sstable::write_components(flat_mutation_reader, unsigned long, seastar::lw_shared_ptr<schema const>, sstables::sstable_writer_config const&, encoding_stats, seastar::io_priority_class const&)::{lambda()#1}::operator()() [clone .cold] at /usr/include/c++/9/ext/new_allocator.h:105
seastar::noncopyable_function<void ()>::direct_vtable_for<seastar::futurize<std::result_of<std::decay<sstables::sstable::write_components(flat_mutation_reader, unsigned long, seastar::lw_shared_ptr<schema const>, sstables::sstable_writer_config const&, encoding_stats, seastar::io_priority_class const&)::{lambda()#1}>::type ()>::type>::type seastar::async<sstables::sstable::write_components(flat_mutation_reader, unsigned long, seastar::lw_shared_ptr<schema const>, sstables::sstable_writer_config const&, encoding_stats, seastar::io_priority_class const&)::{lambda()#1}>(seastar::thread_attributes, std::decay&&, (std::decay<sstables::sstable::write_components(flat_mutation_reader, unsigned long, seastar::lw_shared_ptr<schema const>, sstables::sstable_writer_config const&, encoding_stats, seastar::io_priority_class const&)::{lambda()#1}>::type&&)...)::{lambda(seastar::futurize<std::result_of<std::decay<{lambda()#1}>::type ()>::type> seastar::async<{lambda()#1}>(seastar::futurize<std::result_of<std::decay<{lambda()#1}>::type ()>::type>::type, seastar::thread_attributes, std::decay<{lambda()#1}>::type&&)::work&)#1}::operator()(seastar::futurize<std::result_of<std::decay<{lambda()#1}>::type ()>::type> seastar::async<{lambda()#1}>(seastar::futurize<std::result_of<std::decay<{lambda()#1}>::type ()>::type>::type, seastar::thread_attributes, std::decay<{lambda()#1}>::type&&)::work)::{lambda()#1}>::call(seastar::noncopyable_function<void ()> const*) at /usr/include/c++/9/ext/new_allocator.h:105
seastar::thread_context::main() at /usr/include/c++/9/ext/new_allocator.h:105
?? ??:0
```

Coredump download instruction:

```
gsutil cp gs://upload.scylladb.com/core.scylla.996.a2d470f51fc5489f95c339b9ff52621b.88072.1571926516000000/core.scylla.996.a2d470f51fc5489f95c339b9ff52621b.88072.1571926516000000.gz .
gunzip /var/lib/systemd/coredump/core.scylla.996.a2d470f51fc5489f95c339b9ff52621b.88072.1571926516000000.gz
```

**Comment by bhalevy** [comment]
Do we have binary and logs?

**Comment by juliayakovlev** [comment]
scylla_ami_id: ```ami-02f18b3d30b99fa63```

Logs can be found:

```
23:00:13  +--------------------------------------------------------------------------------------------------------------------------------------------+
23:00:13  |           Collected logs by test-id: 1759fc0b-143e-46dc-bdb3-63efce602484 Directory: /sct/20191023-132728-683564/collected_logs            |
23:00:13  +-------------+------------------------------------------------------------------------------------------------------------------------------+
23:00:13  | Cluster set | Link                                                                                                                         |
23:00:13  +-------------+------------------------------------------------------------------------------------------------------------------------------+
23:00:13  | loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/1759fc0b-143e-46dc-bdb3-63efce602484/20191025_205131/loader-set-1759fc0b.zip  |
23:00:13  | db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/1759fc0b-143e-46dc-bdb3-63efce602484/20191025_205131/db-cluster-1759fc0b.zip  |
23:00:13  | monitor-set | https://cloudius-jenkins-test.s3.amazonaws.com/1759fc0b-143e-46dc-bdb3-63efce602484/20191025_205131/monitor-set-1759fc0b.zip |
23:00:13  | sct-runner  | https://cloudius-jenkins-test.s3.amazonaws.com/1759fc0b-143e-46dc-bdb3-63efce602484/20191025_205131/sct-runner-1759fc0b.zip  |
23:00:13  +-------------+------------------------------------------------------------------------------------------------------------------------------+
```

**Comment by tgrabiec** [comment]
Duplicate of https://github.com/scylladb/scylla/issues/5215, right?

**Comment by tgrabiec** [comment]
This should be fixed by 79d5fed40b034f772445ee02a19e1f51911ffc0c, which removes the call to on_internal_error() from ~mutation_fragment_stream_validator. We see this call in the backtrace here, which proves that the version does not contain the fix.

**Comment by slivne** [comment]
closing the issue - thanks tomasz and julia
